### DISCUSSION TITLE
Question r.e. deprecation of Automatic Schema Matching

### DISCUSSION THREAD

**Discussion by sleepwalker2017** [facebookresearch/hydra#2424] [Q&A]
![image](https://user-images.githubusercontent.com/26128514/196166552-7cbc990d-d783-463a-b3a5-00fa87182e0a.png)

I wrote a yaml file and want to enable type check during runtime.

It seems I can write a data class and register it with the same name with the config file.

I tried it and it worked. But I see the warning.

Will it not be supported in later versions?  How can I use it? 
Or if it will still be supported, the warning seems a bug?

**Comment by sleepwalker2017** [comment]
My code is here.

```python
import hydra
from omegaconf import DictConfig, OmegaConf
import time
from dataclasses import dataclass, field
from hydra.core.config_store import ConfigStore

@dataclass
class InferenceConfig:
    backend: str
    batch_size: int
    input_names: list
    output_name: list = field(default_factory = lambda: list())

@dataclass
class TrainConfig:
    batch_size: int
    data_set: str

@dataclass
class Config:
    device: str
    job_name: str
    inference: InferenceConfig
    train: TrainConfig

cs = ConfigStore.instance()
cs.store(name="config", node=Config)

@hydra.main(config_path="conf", config_name="config", version_base='1.2.0')
def main(cfg: Config) -> None:
    print(cfg)
    print(OmegaConf.to_yaml(cfg))
    with open('result', 'w') as f:
        f.write(OmegaConf.to_yaml(cfg))

if __name__ == "__main__":
    main()
```

**Comment by Jasha10** [comment]
There feature you've identified is called "Automatic Schema Matching".
It is deprecated; see the [docs here](https://hydra.cc/docs/1.2/upgrades/1.0_to_1.1/automatic_schema_matching/) for the deprecation rationale and migration suggestions.

Automatic Schema Matching means that a `ConfigStore` schema with the same name/group as your yaml file will be automatically merged with the contents of your yaml file. The recommend migration is to use a different name (such as `config_schema` instead of `config`) when registering your dataclass in the ConfigStore, and then explicitly include the `config_schema` as an element in the defaults list of your yaml file.
